File: .skills/openclaw-skills/skills/sdk-team/alibabacloud-bailian-videoanalysis/scripts/list_workspace.py
#!/usr/bin/env python3
"""
Query the list of Bailian workspaces.
Uses aliyun modelstudio CLI plugin.
"""

import subprocess
import json
import sys


def main():
    try:
        result = subprocess.run(
            ['aliyun', 'modelstudio', 'list-workspaces', '--user-agent', 'AlibabaCloud-Agent-Skills'],
            capture_output=True,
            text=True,
            timeout=10
        )

        parsed = json.loads(result.stdout)

        workspaces = []
        if 'workspaces' in parsed:
            workspaces = [
                {
                    'workspace_id': ws['workspaceId'],
                    'name': ws['workspaceName']
                }
                for ws in parsed['workspaces']
            ]

        print(json.dumps({'workspaces': workspaces}, indent=2))
    except Exception as error:
        print(json.dumps({
            'error': str(error),
            'recommend': '请确保已安装 aliyun CLI 和 modelstudio 插件（aliyun plugin install --names aliyun-cli-modelstudio）'
        }, indent=2))
        sys.exit(1)


if __name__ == '__main__':
    main()

File: .skills/openclaw-skills/skills/sdk-team/alibabacloud-bailian-rag-knowledgebase/scripts/list_workspace.js
#!/usr/bin/env node
/**
 * Query the list of MaaS workspaces.
 * Uses the Alibaba Cloud default credential chain.
 */

const ModelStudio20260210 = require('@alicloud/modelstudio20260210');
const Util = require('@alicloud/tea-util');
const Credential = require('@alicloud/credentials');
const OpenApi = require('@alicloud/openapi-client');

async function main() {
    let credential = new Credential.default();
    let config = new OpenApi.Config({
      credential: credential,
    });
    config.endpoint = `modelstudio.cn-beijing.aliyuncs.com`;
    let client = new ModelStudio20260210.default(config);
    let listWorkspacesRequest = new ModelStudio20260210.ListWorkspacesRequest({
        maxResults: 50
    });
    let runtime = new Util.RuntimeOptions({
        readTimeout: 8000,
        connectTimeout: 3000
    });
    let headers = {
        "User-Agent": "AlibabaCloud-Agent-Skills/alibabacloud-bailian-rag-knowledgebase"
    };

    try {
        let resp = await client.listWorkspacesWithOptions(listWorkspacesRequest, headers, runtime);
        let statusCode = resp.statusCode;
        if (statusCode == 200) {
            // 输出精简结果，包含工作空间 ID 和名称/描述
            const workspaces = resp.body?.workspaces || [];
            const result = workspaces.map(ws => ({
                workspaceId: ws.workspaceId,
                name: ws.workspaceName
            }));
            console.log(JSON.stringify({
                workspaces: result
            }, null, 2));
        } else {
            console.log(JSON.stringify(resp.body, null, 2))
        }
    } catch (error) {
        console.log(JSON.stringify({
            error: error.message,
            recommend: error.data?.["Recommend"] || ''
        }, null, 2));
        process.exit(1);
    }
}

// 从命令行参数获取 maxResults 和 nextToken (可选)
main();
